intranet/static/js/eighth/schedule.js

```diff
@@ -175,5 +175,8 @@ $(function() {
         if (activities !== "" && !confirm("Are you sure you want to add the following activities without a sponsor?\n" + activities)) {
             e.preventDefault();
         }
+        if($("input[type='checkbox'].unschedule").prop("checked") && $("input[type=checkbox].sticky").prop("checked") > 0 && !confirm("Cancelling a sticky activity allows students to switch out of it. \nDo you want to continue?")){
```

Comment:
First, instead of checking for the `unschedule` checkbox being checked, you should check for the `scheduled` box *not* being checked (but only in a `<tr>` that has the `scheduled` class).

This isn't obvious unless you've used the interface before, but the "unschedule" box is used to completely *delete* a scheduling of an activity. In that case, this dialog does not need to be shown, as all the signup objects will be deleted and users will need to sign up for other activities. *Cancelling* a scheduling of an activity is done by unchecking the main "schedule" box.

Also, even with that change implemented, this current logic will lead to the following undesired behaviors:
1. A confirmation dialog will be shown if *any* scheduling of an activity is being cancelled as long as *any* scheduling of that activity is sticky, even if these flags are set on different schedulings of the activity. You may need to implement logic similar to the `each` loop a few lines up.
2. A confirmation dialog will not be shown if the sticky flag is set on the *activity* (which implicitly applies to all activities). You may need to modify the template to export the `activity.sticky` flag as a Javascript variable in order to easily check for this.

Reply:
Will do.